docs/30DaysOfPWA/advanced-capabilities/07.md

```diff
@@ -4,12 +4,173 @@
+
+- They always use OS- or browser-native UI controls to display the notification. For the web, this is [Notifications API](https://developer.mozilla.org/en-US/docs/Web/API/Notifications_API). It provides the best possible developer and user experience.
+- Users explicitly opt-in to receive push notifications and can opt-out at any time. If the app developer doesn't provide the latter option in the UI, users can use platform native tools to block push notifications for the app (on mobile platforms) or for the origin (on the web).
+- The user device is receiving and showing these notifications regardless of the app is open or not. Thus, for the web platform, we listen and react to the corresponding event in a service worker.
+- There is always a 3rd party service that handles _subscriptions_ and sends push notifications to the user devices. It's called "Messaging Service" and it's a part of either operating system or browser. Messaging Service is a powerful server and network infrastructure able to deliver messages directly to the devices on a huge scale. As a developer, you can't (and don't need to) choose a specific Messaging Service - you implicitly use it in your code that works with Push API during subscription, unsubscription, sending a notification. And it's free to use.
+
+### How to start
+
+Implementing push notifications on the web includes multiple steps: 
+
+1) Generating a set of so-called [VAPID](https://datatracker.ietf.org/doc/html/draft-ietf-webpush-vapid-00) keys to authenticate a user device against the Messaging Service. You need one set of keys (public and private) per app. You can use either [web-push](https://www.npmjs.com/package/web-push) module or one of free online services. Using `web-push`:
+
+```shell
+npm install web-push -g
+web-push generate-vapid-keys --json
+```
+
+2) After the user explicitly opt-in to receive push notifications (for example, by clicking a button in the UI), you use `subscribe()` method of the `PushManager` interface (if it's available) of the active service worker registration:
+
+It is a best practice to either hide or disable all related UI controls if there is no Push API support in the browser:
+
+```javascript
+if (!('PushManager' in window)) {
+
+    // Code to disable or hide push-related UI controls
+
+    console.log('Push API is not supported');
+    return;
+}
+```
+
+"Subscribe" button handler (for the sake of the example, the code is simplified and doesn't contain error handling):
+
+```javascript
+async function subscribeForPush() {
+    const registration = await navigator.serviceWorker.ready;
+    const vapidPublicKey = "publicKey from Step 1";
+    const pushSubscription = await registration.pushManager.subscribe({
+        userVisibleOnly: true, // Should be always true as currently browsers only support explicitly visible notifications
+        applicationServerKey: urlBase64ToUint8Array(vapidPublicKey) // You will find urlBase64ToUint8Array() function in the link below
```

Comment:
It's not immediately clear which link you're talking about. Should we maybe remove this function here to keep the code simple? And then tell readers later that the key needs to be converted, giving them the link.